src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2093,10 +2123,35 @@ private BoundExpression BindArgumentValue(DiagnosticBag diagnostics, ArgumentSyn
             return BindArgumentExpression(diagnostics, argumentSyntax.Expression, refKind, allowArglist);
         }
 
-        private BoundExpression BindOutVariableArgument(DeclarationExpressionSyntax declarationExpression, DiagnosticBag diagnostics)
+        private BoundExpression BindOutVariableArgument(DeclarationExpressionSyntax decl, DiagnosticBag diagnostics)
+        {
+            TypeSyntax typeSyntax = decl.Type;
+            VariableDesignationSyntax designation = decl.Designation;
+            switch (designation.Kind())
+            {
+                case SyntaxKind.DiscardedDesignation:
+                    {
+                        bool isVar;
+                        bool isConst = false;
+                        AliasSymbol alias;
+                        TypeSymbol declType = BindVariableType(designation, diagnostics, typeSyntax, ref isConst, out isVar, out alias);
+                        Debug.Assert(isVar == ((object)declType == null));
+
+                        return new BoundDiscardedExpression((DiscardedDesignationSyntax)designation, declType);
+                    }
+                case SyntaxKind.SingleVariableDesignation:
+                    return BindOutVariableArgument((SingleVariableDesignationSyntax)designation, typeSyntax, decl, diagnostics);
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(designation.Kind());
+            }
+        }
+
+        private BoundExpression BindOutVariableArgument(
+             SingleVariableDesignationSyntax designation,
```

Comment:
>designation [](start = 45, length = 11)

Consider removing this parameter as well. Perhaps rename the function to BindOutVariableDeclarationArgument.